Source/FlavoredNeutrinoContainerInit.cpp

```diff
@@ -69,6 +112,16 @@ namespace
     *Usymmetric = 2. * (amrex::Random()-0.5);
   }
 
+// angular structure as determined by the Minerbo closure
+// Z is a parameter determined by the flux factor
+// mu is the cosine of the angle relative to the flux direction
+// Coefficients set such that the expectation value is 1
+  AMREX_GPU_HOST_DEVICE void minerbo_closure(Real* result, const Real Z, const Real mu){
+	Real minfluxfac = 1e-3;
+	*result = std::exp(Z*mu);
+	if(Z/3.0 > minfluxfac)
+		*result *= Z/std::sinh(Z);
```

Comment:
it seems like the motivation for this is finding C in the Minerbo paper, eq 2.14, so we have the coefficient to multiply `exp(Z*mu)` as in eq 2.13 ...

when we apply the angular factor returned by this closure later on, the code is doing:

```
Real Nnue_thisparticle = parms->st5_nnue*scale_fac * angular_factor;
```

So that is saying, the number of neutrinos in this particle is equal to the number density of neutrinos in its direction `mu` times the volume per particle `scale_fac` -- that makes sense to me so far, so let me trace this out ...

To get the number density of neutrinos in a direction `mu` we have to evaluate eq 2.13 in Minerbo

To get Minerbo's `C` we can use the first eq 2.14 for `m0 = (4*pi*C/Z) * sinh(Z) = 2*pi*int(n(mu),mu) = 4*pi*<n>` where `<n>` is the angle-average number density `parms->st5_nnue` for electrons ...

That leads to `C = <n> * Z / sinh(Z)` and the number density in the direction mu, phi is then:

`n(mu,phi) = C * exp(Z*mu) = <n> * exp(Z*mu) * Z / sinh(Z)`

so the angular factor is:

`n(mu,phi) / <n> = exp(Z*mu) * Z / sinh(Z)`

which matches this function :-) so I assume that's what you did?

Reply:
okay, after sleeping on it, I realized what was bothering me about this ...

as above, saying `parms->st5_nnuQ = <n>` lets you say:

`m0 = (4*pi*C/Z) * sinh(Z) = 2*pi*int(n(mu),mu) = 4*pi*<n>`

so the 4*pi cancels and you get an angular factor:

`n(mu,phi) / <n> = exp(Z*mu) * Z / sinh(Z)`

but a rad-hydro code is likely not calculating `<n>` -- it's probably calculating total neutrino number density `n` 

because `n` is total in all directions, not angle averaged over directions, it's defined by:

`n = int(n(mu,phi) * dmu * dphi)`

so inserting this into `m0` above leads to:

`m0 = (4*pi*C/Z) * sinh(Z) = 2*pi*int(n(mu),mu) = n`

and the angular factor:

`n(mu,phi) / n = exp(Z*mu) * Z / (4 * pi * sinh(Z))`

so whether there is a `4*pi` in the denominator or not depends on what `st5_nnuQ` means and I suspect its a total number density

Reply:
I think there is some confusion about the definition of number density `n(mu,phi)`, since then you have to be careful about factors of 2 and 2pi. I would prefer to think of the angular factor as a quantity that has an expectation value of 1 when sampled uniformly on the sphere. That is, `int(angular_factor dmu dphi) / int(dmu dphi) = 1`. Since the particles are sampled uniformily in direction, the application of the angular factor does not change the number density from the input value. This can also be verified by matching the output number density on the grid with the input value, and those do indeed match.

Reply:
Ah, you're right! Thanks for talking me through this!

I had misunderstood the units for `n(u, phi)` in the Minerbo paper.

Attaching my handwritten notes for future reference.

[Emu-PIC-Angular-Weights.pdf](https://github.com/AMReX-Astro/Emu/files/6387978/Emu-PIC-Angular-Weights.pdf)